ddtrace/tracer/option_test.go

```diff
@@ -249,6 +249,60 @@ func TestServiceName(t *testing.T) {
 	})
 }
 
+//DD_TAGS applicable only
+func TestTagSeparators(t *testing.T) {
+	tags := []struct {
+		in          string
+		expected    map[string]string
+		notExpected []string
+	}{
+		{in: "env:test,aKey:aVal bKey:bVal cKey:", expected: map[string]string{
+			"env": "test", "aKey": "aVal", "bKey": "bVal", "cKey": ""}},
+		{"env:test aKey:aVal     bKey :bVal dKey: dVal cKey:", map[string]string{
+			"env": "test", "aKey": "aVal", "bKey": "", "dKey": "", "dVal": "", "cKey": ""}, []string{"bVal"}},
+		{in: "env :test, aKey : aVal bKey:bVal cKey:", expected: map[string]string{
+			"env": "", "aKey": "", "bKey": "bVal", "cKey": ""}, notExpected: []string{""}},
+		{in: "env:test,aKey:aVal,bKey:bVal,cKey:", expected: map[string]string{
+			"env": "test", "aKey": "aVal", "bKey": "bVal", "cKey": ""}, notExpected: []string{""}},
+		{in: "env:test aKey:aVal bKey:bVal cKey:", expected: map[string]string{
+			"env": "test", "aKey": "aVal", "bKey": "bVal", "cKey": ""}, notExpected: []string{""}},
+		{in: "env:keyWithA:Semicolon bKey:bVal cKey", expected: map[string]string{
+			"env": "keyWithA:Semicolon", "bKey": "bVal", "cKey": ""}, notExpected: []string{""}},
+		{in: "env:keyWith:Lots:Of:Semicolons bKey:bVal", expected: map[string]string{
+			"env": "keyWith:Lots:Of:Semicolons", "bKey": "bVal"}, notExpected: []string{"cKey", ""}},
+		{in: "env:keyWith: Lots:Of:Semicolons ", expected: map[string]string{
+			"env": "keyWith:", "Lots": "Of:Semicolons"}, notExpected: []string{"cKey", ""}},
+		{in: "env:keyWith:  , ,   Lots:Of:Semicolons ", expected: map[string]string{
+			"env": "keyWith:", "Lots": "Of:Semicolons"}, notExpected: []string{"cKey", ""}},
+		{in: "env:keyWith:  , ,   Lots:Of:Semicolons ", expected: map[string]string{
+			"env": "keyWith:", "Lots": "Of:Semicolons"}, notExpected: []string{""}},
+		{in: "a:b,c,d", expected: map[string]string{"a": "b", "c": "", "d": ""}, notExpected: []string{""}},
+		{in: "a:a;", expected: map[string]string{"a": "a;"}, notExpected: []string{}},
+		{in: "a,1", expected: map[string]string{
+			"a": "", "1": ""}, notExpected: []string{}},
+		{in: "key1 :value1  \t key2:  value2", expected: map[string]string{
+			"key1": "", "key2": "", "value2": ""}, notExpected: []string{"value1", ""}},
+		{in: "a:b:c:d", expected: map[string]string{"a": "b:c:d"}, notExpected: []string{""}},
```

Comment:
We should format these one key/val per line to make them readable.
I'm a little concerned about these cases. One is important for backwards-compatibility. The other is important for correctness of the new space-separated pairs.
```
		{
			in:       "a:b c d,e:f g h",
			expected: map[string]string{"a": "b c d", "e": "f g h"},
		},
		{
			in:       "a:b,c,d e:f,g,h",
			expected: map[string]string{"a": "b,c,d", "e": "f,g,h"},
		},
```

Do you think it's feasible to satisfy both of these cases? If not, we can look at some other configuration option and/or assess how big a deal this is.

The main concern is that if a customer upgrades their tracer version (which should be seamless and backwards-compatible) their keys may suddenly change if they have commas or spaces in `DD_TAGS`.

It may be the case that not every case is covered, but we should cover the common cases as much as possible, and have some solution for the cases we can't cover.

Reply:
Readability - got that.

Last example is not actually backwards compatible, that is:
`a:b,c,d e:f,g,h` -> `map[string]string{"a": "b", "c":"", "d":"", "e": "f", "g":"","h":""}` - that would be the result before

I think rules should be clear first, because if we are coming up with complex cases/rules which are not clear for the user, he will be prone to make a mistake in any case.